### PR TITLE
Steve Module Tests

### PR DESCRIPTION
**Pull request by MHDGhanemBalhawan**
Hi,
Please find my test.
Thank you and regards!

### CODE REVIEW

File: module1.js [hunk 1]
```diff
@@ -28,7 +32,11 @@ function question1(name1) {
 */
 
 function question2(number1, number2) {
-  // TODO:
+  if (number1 == number2) {
```
Comment (textbook) [inline module1.js:35]: Remember `==` includes type casts, so e.g. `'1' == 1`; it's generally better to use `===`.

File: module1.js [hunk 2]
```diff
@@ -54,7 +66,14 @@ function question3(number) {
 
 function question4(array) {
   // TODO:
-}
+  var newArr = [];
+  for (var i = 0; i < array.length; i = i + 2){
+    newArr.push(array[i]);
```
Comment (textbook) [inline module1.js:71]: Generally speaking _"every second number"_ would start from the _second_ value (`array[1]`), not the first; the test wasn't quite right, though, so good job getting something approximately right.

File: module1.js [hunk 3]
```diff
@@ -16,7 +16,11 @@
 function question1(name1) {
   if (name1) {
     // TODO:
-    return;
+    var str = name1;
+    var res = str.match(/Code Your Future/g);
+    return true;
```
Comment (textbook) [inline module1.js:21]: Note that you're not actually _using_ `res`, which is a bad sign; your current implementation could be simplified to `return !!name1;`. Also the assignment `var str = name1` is redundant, you can refer to `name1` directly. Finally a regular expression is overcomplicated and actually _breaks_ the required functionality; what if the input was `'A string that mentions Code Your Future but is not right'`?

File: module1.js [hunk 4]
```diff
@@ -80,7 +100,9 @@ function question5(number1, number2) {
 */
 
 function question6(array) {
+
   // TODO:
+  // return `The array contains ${array.length} animals. These are: ${array[]},${element2},${element3},${element4}.`;
```
Comment (textbook) [inline module1.js:105]: This was a good start - think about the array methods you could use to get the second part of the output string.

File: module1.js [hunk 5]
```diff
@@ -111,6 +136,10 @@ function question7(array1, array2, extraElement) {
 
 function question8(retailPrice, taxAmount) {
   // TODO:
+  var ret = retailPrice.toCurrency();
```
Comment (textbook) [inline module1.js:139]: Strings do not have a `toCurrency` method! The error message `TypeError: retailPrice.toCurrency is not a function` tells you where the issue is. You do need to _parse_ the input, though, to extract the numerical part.

File: module1.js [hunk 6]
```diff
@@ -111,6 +136,10 @@ function question7(array1, array2, extraElement) {
...
+  var salesTax = ret * taxAmount;
+  
+  return salesTax;
```
Comment (textbook) [inline module1.js:142]: In general there is no point assigning a variable only to return it in the following line; just `return ret * taxAmount;`.

File: module1.js [hunk 7]
```diff
@@ -96,6 +118,9 @@ function question6(array) {
 
 function question7(array1, array2, extraElement) {
   // TODO:
+  var newArr = array1.concat(array2);
+  newArr.push(extraElement);
+  return newArr;
```
Comment (textbook) [inline module1.js:123]: Nice, this is a good implementation.

File: module1.js [hunk 8]
```diff
@@ -28,7 +32,11 @@ function question1(name1) {
 */
 
 function question2(number1, number2) {
-  // TODO:
+  if (number1 == number2) {
+    return true;
```
Comment (textbook) [inline module1.js:36]: Quite a few other students have also written code like:
```
if (someCondition) {
  return true;
} else {
  return false;
}
```
In these cases note that you can often just `return someCondition;`, or `return !!someCondition;` if it's not already evaluating to `true` and `false`.

**Review by textbook — COMMENTED** [review]
An OK showing overall, 5 answers right out of 8 attempted (when allowing for the test issues on question 4). Did you realise there was a module 2, or did you just run out of time on module 1?

File: module1.js (hunk 1 above)
Comment (reber12) [reply module1.js]: thanks

Comment (MHDGhanemBalhawan) [reply module1.js:35]: I run out of time and I could only take a look on module 2.
Thank you and I'll try to pay more attention next time. Most of the time when I work fast, I miss.

File: module1.js (hunk 2 above)
Comment (MHDGhanemBalhawan) [reply module1.js:71]: What happened was, I checked the test and tried to match the answer.

**Pull request closed without merging**